Creiwry/irl-meeting-app
# Feat redirect signed in user to users index

### Ticket Link
[(Found on Jira in your ticket description)](https://templatethpprojetfinal.atlassian.net/jira/software/projects/THPPROJ/boards/1?selectedIssue=THPPROJ-12)

### User Story
(Linked to on Jira in your ticket description)
https://templatethpprojetfinal.atlassian.net/jira/software/projects/THPPROJ/boards/1?selectedIssue=THPPROJ-12

### What does this PR do?
Modifies UsersController to automatically redirect user to users#index upon successful sign in. 

### Where should the reviewer start?
See next.

### How should this be manually tested?

To test manually, login on root_path and verify that you are redirected to users#index automatically with successful sign in notification (see screenshot).

Regarding the Rspec tests, there were two failures related to the redirect path (see below). However, it seems to me that the tests are written incorrectly? or maybe we changed logic, as we want the user to be redirected automatically to the users#index, not the root path (which is the landing page). Let's check in tomorrow if I've missed something.

Failures:

  1) Devise POST /users/sign_up creates a new user
     Failure/Error: expect(response).to redirect_to(root_path)

       Expected response to be a redirect to <http://www.example.com/> but was a redirect to <http://www.example.com/users>.
       Expected "http://www.example.com/" to be === "http://www.example.com/users".
     # ./spec/requests/devise_request_spec.rb:23:in `block (3 levels) in <main>'

  2) Devise POST /users/sign_in signs in the user
     Failure/Error: expect(response).to redirect_to(root_path)

       Expected response to be a redirect to <http://www.example.com/> but was a redirect to <http://www.example.com/users>.
       Expected "http://www.example.com/" to be === "http://www.example.com/users".
     # ./spec/requests/devise_request_spec.rb:32:in `block (3 levels) in <main>'

Finished in 8.21 seconds (files took 7.01 seconds to load)
102 examples, 2 failures, 28 pending

Failed examples:

rspec ./spec/requests/devise_request_spec.rb:18 # Devise POST /users/sign_up creates a new user
rspec ./spec/requests/devise_request_spec.rb:30 # Devise POST /users/sign_in signs in the user

### Screenshots (if appropriate)
![Screen Shot 2023-09-07 at 23 30 09](https://github.com/Creiwry/irl-meeting-app/assets/137942686/9032add2-c9be-431a-9380-3a42f0974843)

### Are there any breaking changes?
This shouldn't break anything. it's a simple redirect in UsersController.
